x/ccv/provider/keeper/partial_set_security.go

```diff
@@ -67,3 +72,152 @@ func (k Keeper) HandleOptOut(ctx sdk.Context, chainID string, providerAddr types
 
 	return nil
 }
+
+// ComputeValidatorUpdates computes the validator updates needed to be sent to the consumer chain to capture
+// the newly opted-in and opted-out validators, as well as validators that unbonded.
+func (k Keeper) ComputeValidatorUpdates(ctx sdk.Context,
```

Comment:
I feel like I am missing context so my judgement may not be accurate, but it feels like one of ComputeValidatorUpdates and ComputeNextValidators is unnecessary?
Wouldn't NextValidators + a diffing method like Jehan suggested a bit ago do the trick?

Reply:
See my previous comment [here](https://github.com/cosmos/interchain-security/pull/1642#discussion_r1492109174).
More specific to the potentiality of a diff method, I guess you could first call `ComputeNextValidators` and then call `diff(currentValidators, nextValidators)` but this `diff` method would be equally complex `ComputeValidatorUpdates` and would have a different method signature compared to `ComputeNextValidators`. So, I don't see much the advantage to this.
Does that sound reasonable?

Reply:
Yeah, I agree. I think I would still find the diff method more elegant, but it probably comes down to personal preference, and at the end of the day both are ok.